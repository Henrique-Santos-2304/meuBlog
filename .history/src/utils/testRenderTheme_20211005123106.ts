import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
