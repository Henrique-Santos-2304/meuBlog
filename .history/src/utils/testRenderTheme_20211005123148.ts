import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
