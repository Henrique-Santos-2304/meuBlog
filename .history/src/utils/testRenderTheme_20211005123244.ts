import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
