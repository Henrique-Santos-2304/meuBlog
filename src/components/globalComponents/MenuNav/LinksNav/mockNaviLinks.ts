export const mocknavLinks = {
  navigations: [
    {
      title: "Inicio",
      url: "#",
      text: "Inicio",
    },
    {
      title: "Projetos",
      url: "#projects",
      text: "Projetos",
    },
    {
      title: "Habilidades",
      url: "#skilss",
      text: "Habilidades",
    },
    {
      title: "Contato",
      url: "#contact",
      text: "Contato",
    },
  ],
};
