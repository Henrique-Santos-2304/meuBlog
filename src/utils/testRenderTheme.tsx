import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
