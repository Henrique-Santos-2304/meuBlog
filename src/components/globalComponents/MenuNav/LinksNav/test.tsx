import LinkNav from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";
import { mocknavLinks } from "./mockNaviLinks";

describe("<LinkNav />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<LinkNav {...mocknavLinks} />);
    expect(container).toMatchSnapshot();
  });
  it("should have four links the navigation ", () => {
    renderTheme(<LinkNav {...mocknavLinks} />);
    const linksNavigation = screen.getAllByRole("link");
    expect(linksNavigation).toHaveLength(4);
  });
});
