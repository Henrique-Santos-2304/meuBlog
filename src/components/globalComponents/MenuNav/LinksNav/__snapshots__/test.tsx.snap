// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkNav /> should render  1`] = `
.c0 {
  color: #f8f8f8;
  font-size: 1.6rem;
  margin: 1.5rem 0;
  position: relative;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

.c0:hover::before {
  content: "";
  position: absolute;
  bottom: -1rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 120%;
  height: 0.3rem;
  background: #486afe;
  -webkit-animation: fSVfHE 0.6s ease-in-out;
  animation: fSVfHE 0.6s ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/#"
    >
      Inicio
    </a>
  </div>
  <div
    class="c0"
  >
    <a
      href="/#projects"
    >
      Projetos
    </a>
  </div>
  <div
    class="c0"
  >
    <a
      href="/#skilss"
    >
      Habilidades
    </a>
  </div>
  <div
    class="c0"
  >
    <a
      href="/#contacts"
    >
      Contato
    </a>
  </div>
</div>
`;
